src/lib.rs

```diff
@@ -715,7 +705,34 @@ impl CommonMarkViewerInternal {
                 ui.add(egui::Separator::default().horizontal());
             }
             pulldown_cmark::Tag::CodeBlock(_) => {
-                self.fenced_code_block = None;
+                if let (Some(lang), Some(text), Some(max_width)) = (
+                    self.fenced_code_block_lang.clone(),
+                    self.fenced_code_block_contents.clone(),
```

Comment:
Maybe change these to use [Option::take](https://doc.rust-lang.org/std/option/enum.Option.html#method.take). Should avoid both the cloning and the need for setting them to `None` afterwards.